ToDoList/src/ToDoList.DependencyInjection/Container.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+
+using Unity;
+using Unity.Injection;
+using Unity.Lifetime;
+
+namespace ToDoList.DependencyInjection
+{
+    public class Container
+    {
+        private readonly UnityContainer _container = new UnityContainer();
+
+        public IUnityContainer GetContainer()
+        {
+            return _container;
+        }
+
+        public void RegisterType<T>(Func<T> injectionFunction)
+            => _container.RegisterType<T>(new InjectionFactory(_ => injectionFunction()));
+
+        public void RegisterType<TFrom, TTo>(LifetimeManager managerType)
+            where TTo: TFrom
+        {
+            switch (managerType)
+            {
+                case LifetimeManager.Transient:
+                    _container.RegisterType<TFrom, TTo>(new TransientLifetimeManager());
+                    break;
+                case LifetimeManager.Hierarchical:
+                    _container.RegisterType<TFrom, TTo>(new HierarchicalLifetimeManager());
+                    break;
+                default:
```

Comment:
Consider stating the `LifetimeManager.Singleton` explicitly and throwing an exception on `default` (this will occur when a new value gets added to the enum yet this method is not updated) – that is considered a best practice (in C# and in JS as well)…

Reply:
It would also be nice to have tests for the switch, but that is kind of nitpicking of me (yet if you used TDD, you would end up with some, so I at least felt the urge to mention them).